art/attacks/evasion/steal_now_attack_latter.py

```diff
@@ -0,0 +1,747 @@
+        cur_mask = torch.cat(cur_mask, dim=-1)
+
+        tile = torch.cat([tile, cur_tile], dim=0)
+        mask = torch.cat([mask, cur_mask], dim=0)
+
+    return tile, mask
+
+# internal used
+class TileObj:
+    def __init__(self,
+                 tile_size: int,
+                 device: "torch.cuda.device") -> None:
+        import torch
+        self.patch = torch.zeros((3, tile_size, tile_size), device = device)
+        self.diff = torch.ones([], device = device) * self.patch.shape.numel()
+        self.bcount = 0
+        self.eligible = False
+
+    def update(self,
+               eligible = None,
+               bcount = None,
+               diff = None,
+               patch = None
+               ) -> None:
+
+        if not (eligible is None):
+            self.eligible = eligible
+
+        if not (bcount is None):
+            self.bcount = bcount
+
+        if not (diff is None):
+            self.diff = diff
+
+        if not (patch is None):
+            self.patch = patch
+
+    def compare(self, target: "TileObj") -> bool:
+        if self.eligible == True and target.eligible == False:
+            return True
+        elif self.eligible == False and target.eligible == True:
+            return False
+        else:
+            if self.bcount > target.bcount:
+                return True
+            elif self.bcount < target.bcount:
+                return False
+            else:
+                if self.diff < target.diff:
+                    return True
+                else:
+                    return False
+
+
+class TileArray:
+    def __init__(self,
+                 xyxy: list, 
+                 threshold: int, 
+                 tile_size: int, 
+                 k: int, 
+                 device: "torch.cuda.device") -> None:
+        import torch
+        self.threshold = threshold
+        self.strides = tile_size
+        self.device = device
+        self.xyxy = torch.Tensor(xyxy).to(device)
+        self.k = k
+        self.patch_list = [TileObj(tile_size=tile_size, device=device)] * self.k
+
+    def insert(self, target: TileObj) -> None:
+
+        if target.bcount < self.threshold:
+            return
+
+        prev = self.patch_list
+        out = []
+        for k_it in range(self.k):
+            if target.compare(prev[k_it]):
+                out.append(target)
+                out = out + prev[k_it:]
+                break
+            else:
+                out.append(prev[k_it])
+
+        self.patch_list = out[:self.k]
+
+    def pop(self) -> None:
+        out = self.patch_list[1:] + [TileObj(tile_size=self.strides, device=self.device)]
+        self.patch_list = out
+
+
+class SNAL(EvasionAttack):
+    """
+    Steal Now and Attack Later
+    | Paper link: https://arxiv.org/abs/2404.15881
+    """
+
+    attack_params = EvasionAttack.attack_params + [
+        "eps",
+        "max_iter",
+        "num_grid",
+        "batch_size",
+    ]
+
+    _estimator_requirements = ()
```

Comment:
Does this attack work for all estimators or only object detection estimators in PyTorch? If yes, add
```suggestion
    _estimator_requirements = (PyTorchObjectDetector)
```